cEP-0002.md

```diff
@@ -69,41 +69,40 @@ backwards compatibility. There will be new versions of `LocalBear` and
 `GlobalBear` that will be called `FileWiseBear` and `ProjectWideBear`
 respectively.
 
-# The File Proxy
+# The File Factory
 
-With the new bear design we are also moving to a new wrapper object to wrap
-every file. The file proxy is an object that contains properties about the
-file currently processed (such as the filename). It contains:
+With the new bear design we are also moving to a new factory pattern
+for loading files.
 
-+ The filename, this is the starting point for every other attribute.
-+ The content of the file as a string.
-+ The content of the file, line-splitted.
+The factory instance is located at `coalib.input.file.factory`,
+and will default to an instance of a class `LineCacheFileFactory` 
+that uses `linecache` to provide access to file contents.
 
-Note: Other properties can be included later as and when required.
+Additional FileFactory implementations will be created as required
```

Comment:
I'm a little confused here. I'm guessing my LineCacheFileFactory you mean you can access it line by line or it'd be a list of lines ?
The other instances would be things like WordCacheFileFactory which would be word-by-word or LetterCacheFileFactory  and so on?

Can this be done neater ? like using a argument which specifies the single unit ? Eg: `FileFactory(atom='word')`

Is there any other use for additional implementations ?